tools/webdriver/webdriver/bidi/transport.py

```diff
@@ -70,10 +70,14 @@ async def end(self) -> None:
 
     async def read_messages(self) -> None:
         assert self.connection is not None
-        async for msg in self.connection:
-            if not isinstance(msg, str):
-                raise ValueError("Got a binary message")
-            await self.handle(msg)
+        try:
+            async for msg in self.connection:
+                if not isinstance(msg, str):
+                    raise ValueError("Got a binary message")
+                await self.handle(msg)
+        except websockets.exceptions.ConnectionClosed:
```

Comment:
Per the test failures `websockets.exceptions` can't be used like this. You might need to import the specific module.